### PR TITLE
New component textArea

### PR DESCRIPTION
**Pull request by aprentout**
### What does this PR do?

Related to: [ENG-269](https://linear.app/upfluence/issue/ENG-269/[groundwork]-new-osstextarea-component)

### What are the observable changes?
Capture d’écran 2023-06-13 à 09 23 27
Capture d’écran 2023-06-13 à 09 23 37

### Good PR checklist
- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/o-s-s/text-area.hbs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<div class="fx-1">
```
Comment (phndiaye) [inline addon/components/o-s-s/text-area.hbs:1]: can you add a `...attributes` here?

File: addon/components/o-s-s/text-area.hbs [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  <div class="oss-textarea-container {{if @errorMessage ' oss-textarea-container--errored'}}">
+    <Textarea {{on "change" (fn this._onChange @value)}} @value={{@value}} placeholder={{@placeholder}}
+              disabled={{@disabled}} class="oss-textarea {{this.computedClass}}" rows={{this.rows}} ...attributes />
```
Comment (phndiaye) [inline addon/components/o-s-s/text-area.hbs:4]: having the `...attributes` here might cause some issues when targeting the component imo

File: addon/components/o-s-s/text-area.hbs [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  </div>
+  {{#if @errorMessage}}
+    <span class="text-color-error margin-top-xxx-sm">{{@errorMessage}}</span>
```
Comment (phndiaye) [inline addon/components/o-s-s/text-area.hbs:7]: Classes:
- `text-color-error` -> `font-color-error-500`
- `margin-top-xxx-sm` -> `margin-top-px-6`

File: addon/components/o-s-s/text-area.stories.js
Comment (phndiaye) [inline addon/components/o-s-s/text-area.stories.js:1]: Mentions of "input" can be replaced w/ Textarea

File: addon/components/o-s-s/text-area.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    if (this.args.resize) {
+      assert(
+        '[component][OSS::TextArea] The @resize parameter should be a value of resize',
+        ['vertical', 'horizontal', 'both', 'none'].includes(this.args.resize)
+      );
+    }
+  }
+
+  get rows(): number {
+    return this.args.rows || 1;
```
Comment (JulienVannier66) [inline addon/components/o-s-s/text-area.ts:28]:
```suggestion
    return this.args.rows || 2;
```
Maybe set the same default value that define in [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea) for row

File: addon/components/o-s-s/text-area.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+
+interface OSSTextAreaArgs {
+  rows?: number;
+  resize?: 'vertical' | 'horizontal' | 'both' | 'none';
```
Comment (JulienVannier66) [inline addon/components/o-s-s/text-area.ts:7]: What happen when we use `both` as `resize`, I don't anything here. Add the case in `computedClass` method no?

File: addon/components/o-s-s/text-area.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    else if (this.args.resize === 'none') classes.push('oss-textarea--resize-none');
+
+    return classes.join(' ');
+  }
+
+  @action
+  _onChange(value: string): void {
+    if (this.args.onChange) {
+      this.args.onChange(value);
+    }
```
Comment (JulienVannier66) [inline addon/components/o-s-s/text-area.ts:44]:
```suggestion
  this.args.onChange?.(value);
```
You can simplify that :smile:

File: tests/integration/components/o-s-s/text-area-test.ts [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    });
+
+    test('When resize is none should have class resize-none', async function (assert) {
+      this.resize = 'none';
+      await render(hbs`<OSS::TextArea @resize={{this.resize}}/>`);
+      assert.dom(this.textareaSelector).hasNoClass('oss-textarea--resize-v');
+      assert.dom(this.textareaSelector).hasNoClass('oss-textarea--resize-h');
+      assert.dom(this.textareaSelector).hasClass('oss-textarea--resize-none');
+    });
+  });
```
Comment (JulienVannier66) [inline tests/integration/components/o-s-s/text-area-test.ts:52]: Missing the test for `this.resize = 'both';`

File: addon/components/o-s-s/text-area.stories.js
Comment (Miexil) [reply addon/components/o-s-s/text-area.stories.js:1]: Agreed, but it's still an input though :trollface:

File: addon/components/o-s-s/text-area.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  get rows(): number {
+    return this.args.rows || 2;
+  }
+
+  get computedClass(): string {
+    const classes: string[] = [];
+    if (this.args.resize === 'vertical') classes.push('oss-textarea--resize-v');
+    else if (this.args.resize === 'horizontal') classes.push('oss-textarea--resize-h');
+    else if (this.args.resize === 'none') classes.push('oss-textarea--resize-none');
+
```
Comment (Miexil) [inline addon/components/o-s-s/text-area.ts:36]:
```WillyWonka
else if (this.args.resize === 'both') classes.push('oss-textarea--resize-both');
```

```less
  &--resize-both {
    resize: both;
  }
```

**Review by Miexil — APPROVED** [review]

**Review by JulienVannier66 — APPROVED** [review]

**Pull request merged**